### PR TITLE
Add document about Create a new cop

### PR DESCRIPTION
**Pull request by ydah**
Resolve: https://github.com/rubocop/rubocop-rspec/issues/1887#issuecomment-2123486142

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [-] Added tests.
- [x] Updated documentation.
- [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: docs/modules/ROOT/pages/development.adoc [hunk 1]
```diff
@@ -2,6 +2,48 @@
...
+[create] lib/rubocop/cop/rspec/cop_name.rb
+[create] spec/rubocop/cop/rspec/cop_name_spec.rb
+[modify] lib/rubocop/cop/rspec_cops.rb - `require_relative 'rspec/cop_name'` was injected.
+[modify] A configuration for the cop is added into config/default.yml.
+Do 4 steps:
+  1. Modify the description of RSpec/CopName in config/default.yml
+  2. Implement your new cop in the generated file!
+  3. Commit your new cop with a message such as
+     e.g. "Add new `RSpec/CopName` cop"
+  4. Run `bundle exec rake changelog:new` to generate a changelog entry
```
Comment (pirj) [inline docs/modules/ROOT/pages/development.adoc:24]: This doesn’t apply

File: docs/modules/ROOT/pages/development.adoc [hunk 2]
```diff
@@ -2,6 +2,48 @@
...
+* The name starts with a noun instead of a verb, e.g. `ArrayAlignment` instead of `AlignArray`
+* The name is easy to understand, e.g. `IndentationStyle` instead of just `Tab`
+* The name is specific, e.g. `DuplicateHashKey` instead of just `DuplicateKey`
+* The name is neutral when possible and accommodates multiple styles when feasible, e.g. `EmptyLineBeforeBegin`.
+* The name uses commonly-used terms, e.g. `RedundantPercentI` instead of `RedundantPercentSymbolArray`
+* The name uses correct terms, e.g. arguments in a method call, and parameters in a method signature
+* Lines with no symbols are called "empty", not "blank", e.g. `LeadingEmptyLines` instead of `LeadingBlankLines`
+* Prefer "redundant" to "unneeded", e.g. `RedundantSelf` instead of `UnneededSelf`
+
+See the https://github.com/rubocop/rubocop-rspec/blob/master/config/obsoletion.yml["renamed" section of `config/obsoletion.yml`]
```
Comment (pirj) [inline docs/modules/ROOT/pages/development.adoc:44]: This obsoletion list will eventually be reset in RuboCop 2. Should we pin to some gitref?

**Review by pirj — APPROVED** [review]
Thanks!

File: docs/modules/ROOT/pages/development.adoc (hunk 1 above)
Comment (ydah) [reply docs/modules/ROOT/pages/development.adoc:24]: Oh.... This is a copy and paste of the log output when the new_cop command is executed with rubocop-rspec 😅

Comment (ydah) [reply docs/modules/ROOT/pages/development.adoc]: I updated it.

File: docs/modules/ROOT/pages/development.adoc (hunk 2 above)
Comment (ydah) [reply docs/modules/ROOT/pages/development.adoc]: Indeed it is. I updated it.

**Review by pirj — APPROVED** [review]
Nice! Thanks

**Pull request merged**